rpm_spec/gui-agent.spec.in

```diff
@@ -126,6 +124,10 @@ fi
 %posttrans
     /usr/bin/glib-compile-schemas %{_datadir}/glib-2.0/schemas &> /dev/null || :
 
+if ! [ "$(ls -A /etc/init.d/ 2> /dev/null)" ]; then
+    rm -rf /etc/init.d
```

Comment:
Since the intention is to remove empty directory, it should be `rmdir`. Otherwise you could accidentally remove something else (symlink for example). Also, check if directory exists, before checking for it emptiness.